### PR TITLE
Unify storage measurement unit multipliers to fractional binary, instead of decimal

### PR DESCRIPTION
**Pull request by paplorinc**
Fixes https://github.com/gradle/gradle/issues/14540

Ended up using `BigDecimal` for greater precision - `double` had measurable rounding errors around big transitions.

Also unified the fractional part to be displayed in both cases - validated against https://www.ibm.com/support/knowledgecenter/SSQRB8/com.ibm.spectrum.si.doc/fqz0_r_units_measurement_data.html#fqz0_r_data_storage_values__percentage_diff_between_decimal_binary.

Decided to always round down, even though the `IEEE 754R Decimal64` format uses `HALF_EVEN` - we don't need to avoid rounding biases here, so truncation is more predictable.

### CODE REVIEW

File: subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java [hunk 1]
```diff
@@ -42,33 +52,33 @@ public static int percentOf(long fraction, long total) {
...
-        char pre = "kMGTPE".charAt(exp - 1);
-        return format("%.1f %sB", bytes / Math.pow(unit, exp), pre);
     }
 
     /**
      * gets ordinal String representation of given value (e.g. 1 -> 1st, 12 -> 12th, 22 -> 22nd, etc.)
      */
     public static String ordinal(int value) {
-        String[] sufixes = new String[]{"th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th"};
+        String[] suffixes = new String[]{"th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th"};
```
Comment (lptr) [inline subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java:74]: This should be a constant.

File: subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java [hunk 2]
```diff
@@ -16,12 +16,23 @@
...
 /**
  * Utility methods for working with numbers
  */
 public class NumberUtil {
+    public static final int BASE_LOG2 = 10;
+    public static final int KIB_BASE = 1 << BASE_LOG2;
+    private static final int FRACTIONAL_DIGIT_COUNT = 1;
+    private static final MathContext MC = new MathContext(String.valueOf(KIB_BASE).length() + FRACTIONAL_DIGIT_COUNT, FLOOR);
+
+    private static final String[] UNITS = new String[]{" B", " KiB", " MiB", " GiB", " TiB", " PiB", " EiB"};
```
Comment (lptr) [inline subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java:34]: Where did we decide to go with `KiB` instead of `kB` etc.?

Comment (paplorinc) [reply subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java:34]: Please see the related issue

File: subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java (hunk 1 above)
Comment (paplorinc) [reply subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java]: Thanks, done

File: subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java [hunk 3]
```diff
@@ -16,12 +16,24 @@
...
+    private static final String[] ORDINAL_SUFFIXES = new String[]{"th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th"};
```
Comment (marcphilipp) [inline subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java:35]: I always feel a bit uneasy when I see arrays used as constants. Can we use `ImmutableLists` instead?

Comment (paplorinc) [reply subprojects/base-services/src/main/java/org/gradle/internal/util/NumberUtil.java]: Done

**Pull request merged**